### DISCUSSION TITLE
Return value must be of type Filament\Support\Enums\Alignment|string, resource returned

### DISCUSSION THREAD

**Discussion by Kovah** [filamentphp/filament#8973] [Help] [answered]
Hi all,

**Edit**: this also affects all other plugins and forms now. The same happens with the Curator plugin I use and with custom forms for resources I created.

I upgraded to Filament 3 a while ago, but since then the Spatie Settings plugin seems to be broken. I cannot open any settings pages anymore. When I upgraded to Filament 3, I also changed the settings pages according to [the plugin docs](https://filamentphp.com/plugins/filament-spatie-settings).

It looks like this:
```php
namespace App\Filament\Pages;

use App\Settings\NewsletterSettings;
use Filament\Forms;
use Filament\Forms\Components\Fieldset;
use Filament\Forms\Components\TextInput;
use Filament\Forms\Form;
use Filament\Pages\SettingsPage;

class NewsletterSettings extends SettingsPage
{
    protected static ?string $navigationIcon = 'heroicon-o-cog-6-tooth';

    protected static string $settings = NesletterSettings::class;

    public function form(Form $form): Form
    {
        return $form
            ->schema([
                Fieldset::make('List IDs')->schema([
                    TextInput::make('list_id_all_users')
                        ->integer()
                        ->label('All users')
                        ->required(),
                    // ...
                ])
            ]);
    }
}
```
The pages work when I open them locally, but when deployed to production, I get the following error when opening the pages:
```
Filament\Pages\BasePage::getFormActionsAlignment(): 
Return value must be of type Filament\Support\Enums\Alignment|string, resource returned 
(View: /app/vendor/filament/filament/resources/views/components/form/actions.blade.php) 
(View: /app/vendor/filament/filament/resources/views/components/form/actions.blade.php)
```

Not sure what the issue is. When the app is deployed to production, I run the steps suggested [in the docs](https://filamentphp.com/docs/3.x/panels/installation#deploying-to-production) directly after running `composer install`, including running `php artisan filament:upgrade`.

**Comment by Kovah** [comment]
I digged deeper into this because it also affects other plugins and has now made Filament completely unusable.

The root cause seems to be the cached views. The error actually comes from a file like `storage/framework/views/f65a768c985d1593409aafe991f49d6f.php` where I find the following content:
```php
<?php if (isset($component)) { $__componentOriginal71c6471fa76ce19017edc287b6f4508c = $component; } ?>
<?php $component = Illuminate\View\AnonymousComponent::resolve([
  'view' => 'filament-actions::components.actions',
  'data' => [
    'actions' => $actions,
    'alignment' => $alignment ?? $this->getFormActionsAlignment(),
    'fullWidth' = $fullWidth
  ]] + (isset($attributes) && $attributes instanceof Illuminate\View\ComponentAttributeBag ? (array) $attributes->getIterator() : [])); ?>
<?php $component->withName('filament-actions::actions'); ?>
```

**Reply by VincvanAs** [reply]
There is a typo in your     `protected static string $settings = NesletterSettings::class;`

**Reply by Kovah** [reply]
Thanks for the hint, but that's not the issue. I renamed some classes due to privacy concerns.

**Reply by VincvanAs** [reply]
Well tbh im just shooting blindly here. But as far as i can tell ->integer() is not valid anymore and should be replaced by ->numeric()?  In v3 you can define ->alignLeft(), ->alignRight() and center. Somehow you have defined something on several form fields that is calling this method but is returned as a string.  You could try to define ->alignCenter() on your form fields to see if this is solving your issue.

**Comment by Kovah** [comment]
Another update...

This goes deeper than I would like to. I edited `vendor/filament/filament/src/Pages/BasePage.php` to allow returning anything to have a chance taking a look what is wrong with this $alignment variable. The next errors came from a compiled version of the view `/app/vendor/filament/actions/resources/views/components/actions.blade.php`:

```php
    <?php
        $actions = array_filter(
            $actions,
            fn ($action): bool => $action->isVisible(),
        );
        if (! $alignment instanceof Alignment) {
            $alignment = Alignment::tryFrom($alignment) ?? $alignment; // <<<
        }
    ?>
```
now throws
```
Filament\Support\Enums\Alignment::tryFrom(): Argument #1 ($value) must be of type string|int, resource given
```

Adding a simple `var_dump($alignment);exit;` results in a segmentation fault:
```
[06-Nov-2023 16:19:44] WARNING: [pool www] child 101 exited on signal 11 (SIGSEGV - core dumped) after 174.064654 seconds from start
[06-Nov-2023 16:27:54] WARNING: [pool www] child 141 exited on signal 11 (SIGSEGV - core dumped) after 520.986733 seconds from start
[06-Nov-2023 16:29:04] WARNING: [pool www] child 146 exited on signal 11 (SIGSEGV - core dumped) after 559.614416 seconds from start
[06-Nov-2023 16:29:54] WARNING: [pool www] child 86 exited on signal 11 (SIGSEGV - core dumped) after 838.458392 seconds from start
[06-Nov-2023 16:30:53] WARNING: [pool www] child 333 exited on signal 11 (SIGSEGV - core dumped) after 178.829159 seconds from start
```

I'm really lost now. Why the hell is $alignment a resource and where does it come from?

**Reply by Kovah** [reply, marked as answer]
I figured out that - for whatever reason - the Opcache preloading caused this to happen. No idea why. I turned it off to check if something happens, and now everything is working. 🤷
